### PR TITLE
feat(cli): report typescript version via `prisma --version`

### PR DESCRIPTION
**Pull request by FGoessler**
This PR:
- adds the TypeScript version to the `prisma --version` output
- closes https://github.com/prisma/team-orm/issues/1371

It also handles corner cases for Bun and Deno, although I did not add explicit test cases for this. I am not aware of test infrastructure for Deno and Bun to cover this easily either.

### CODE REVIEW

File: packages/internals/src/cli/getTypescriptVersion.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  try {
+    // On node.js this will import the project defined typescript package.
+    // On bun this will import the bun bundled typescript package.
+    return (await import('typescript')).default.version
+  } catch (_) {
+    // On deno the dynamic import of typescript will fail but typescript is natively available and its version is available in process.versions.
+    // If none of this worked typescript is likely not installed / available.
+    return (await getProcessObject())?.versions.typescript || 'unknown'
+  }
+}
```
Comment (jkomyno) [inline packages/internals/src/cli/getTypescriptVersion.ts:22]: Can we use [subpath imports](https://nodejs.org/api/packages.html#subpath-imports) instead, and have different `getTypeScriptVersion.[].ts` files for each major runtime that behaves differently? You can refer to [1](https://jkomyno-armada-js-2024.vercel.app/25?clicks=1), [2](https://jkomyno-armada-js-2024.vercel.app/26) for additional resources.

E.g., you may have a `getTypeScriptVersion.deno.ts` for Deno only, and a `getTypeScriptVersion.ts` for Node.js & Bun. Ideally, you should also import `node:process` in both files, regardless of whether `process` exist as a global variable.

Comment (FGoessler) [reply packages/internals/src/cli/getTypescriptVersion.ts:22]: Just tried this out but had no success in verifying that it actually works as our whole sandbox infrastructure is fitted for node.js and am not able right now to run `prisma --version` via deno based on my local build.

I also feel like using subpath imports here might be a bit of an overkill and actually spreads complexity around from the single `getTypeScriptVersion.ts` file into the `package.json` and maybe even further into the build and import system. But it's also the first time I am using subpath imports myself so I likely not have the full picture.

Comment (jkomyno) [reply packages/internals/src/cli/getTypescriptVersion.ts]: Alright, thanks for trying. Let's then avoid using subpath imports (we already use them in `@prisma/client`, but in fact, it'd be the first time we'd use them within `@prisma/internals`.

**Review by jkomyno — APPROVED** [review]
